### PR TITLE
Keywords unlocked 2

### PR DESCRIPTION
**Pull request by bramtayl**
Continuation of #25647, with keyword arguments which needed renaming as suggested here #25188. I chose `ncores` instead of `cores` following the discussion in #25659.

### CODE REVIEW

File: base/interactiveutil.jl [hunk 1]
```diff
@@ -580,22 +580,22 @@ end
 
 # `methodswith` -- shows a list of methods using the type given
 """
-    methodswith(typ[, module or function][, showparents::Bool=false])
+    methodswith(typ[, module or function]; parents::Bool=false])
```
Comment (JeffBezanson) [inline base/interactiveutil.jl]: I think this should be called `supertypes`, following e.g. the `supertype` function.

File: base/reducedim.jl [hunk 2]
```diff
@@ -667,7 +667,7 @@ function findminmax!(f, Rval, Rind, A::AbstractArray{T,N}) where {T,N}
 end
 
 """
-    findmin!(rval, rind, A, [init=true]) -> (minval, index)
+    findmin!(rval, rind, A) -> (minval, index)
```
Comment (nalimilan) [inline base/reducedim.jl]: The `init` argument exists. I guess you intended to use `;` to show it's a keyword argument? Same below.

File: base/reflection.jl [hunk 3]
```diff
@@ -672,7 +672,7 @@ function signature_type(@nospecialize(f), @nospecialize(args))
 end
 
 """
-    code_lowered(f, types, expand_generated = true)
+    code_lowered(f, types; expand_generated = true)
```
Comment (nalimilan) [inline base/reflection.jl]: Maybe this one could just be `generated`? That would parallel `if @generated`.

File: base/reducedim.jl (hunk 2 above)
Comment (bramtayl) [reply base/reducedim.jl]: No I did this on purpose. https://github.com/JuliaLang/julia/issues/25188#issuecomment-353028482 where Stefan speculated init was a private keyword.

File: base/reflection.jl (hunk 3 above)
Comment (bramtayl) [reply base/reflection.jl]: I commented about this here: https://github.com/JuliaLang/julia/issues/25188#issuecomment-359352314. I'm not too passionate though so if people feel like `generated` makes sense I can change it.

File: base/reducedim.jl (hunk 2 above)
Comment (nalimilan) [reply base/reducedim.jl]: Sorry, I had missed that. I'd remove the argument completely then, given that it doesn't seem to be used internally.

FWIW, `sortperm!` has an `initialized=false` argument which is somewhat similar (but reversed).

Comment (bramtayl) [reply base/reducedim.jl]: Looks like `init` traces back to #6894. @timholy is there any reason to keep `init`?

**Comment by JeffBezanson** [comment]
I'll fix this up.

**Pull request closed without merging**